pkg/operator/staticpod/startupmonitor/fallback.go

```diff
@@ -128,7 +136,49 @@ func (f *staticPodFallback) fallbackToPreviousRevision(reason, message string) e
 }
 
 func (f *staticPodFallback) markRevisionGood() error {
-	return f.createLastKnowGoodRevisionFor(f.revision, true)
+	if err := f.createLastKnowGoodRevisionFor(f.revision, true); err != nil {
+		return err
+	}
+
+	if f.operatorClient == nil || len(f.nodeName) == 0 {
+		return nil
+	}
+
+	// the startup-monitor is authorative to signal readiness of a revision. The operator
+	// is waiting for this to happen. Otherwise, the operator could assume readiness while
+	// the startup-monitor falls back, leading to an awkward situation.
+	// Note that this will retry forever, with a backoff.
+	return retry.OnError(retry.DefaultRetry, func(error) bool { return true }, func() error {
+		_, status, rv, err := f.operatorClient.GetStaticPodOperatorState()
```

Comment:
will `status` come from a live GET or from the cache ?

Reply:
live get. It has to. Chance is high that it works on first try.